pkg/controllers/machine_controller_nics.go

```diff
@@ -216,7 +216,6 @@ func (r *MachineReconciler) reconcileDesiredNetworkInterface(
 	}
 
 	mountedNic, ok := mountedNics[nic.Name]
-	mountedNic.networkInterface.Handle = providerNic.Handle
 	if ok && reflect.DeepEqual(mountedNic.networkInterface, providerNic) {
```

Comment:
Afai remember, this line was added while adapting the controllers to the new APINet resources with Axel, to prevent the DeepEqual from failing. 
This issue emerged as the `.Handle` was newly defined with APINet resource adaptations from the Plugin, but it is not set in the `MountedNic` structure. 
I would recommend testing the flow with this change, and if it's still failing, maybe have a more holistic solution.

Reply:
Sure, thanks.